tsfc/spectral.py

```diff
@@ -44,14 +32,86 @@ def Integrals(expressions, quadrature_multiindex, argument_multiindices, paramet
 
     :returns: list of integral representations
     """
+    # Rewrite: a / b => a * (1 / b)
+    expressions = replace_division(expressions)
```

Comment:
As far as I remember @tj-sun porting the old COFFEE code, the logic was something like this: apply this rewriting if `b` is a number or a `Symbol`. A `Symbol` could either mean a constant or a variable reference. Generally the denominator was assigned to a temporary if it was used more than once. That is, instead of dividing `n` times with the same number, the substitution makes 1 division and `n` multiplication. Most common use case is the calculation of the inverse Jacobian, where the determinant is the denominator.